rust/xaynet-server/src/storage/impls.rs

```diff
@@ -206,21 +208,115 @@ impl FromRedisValue for AddSumParticipant {
     }
 }
 
-#[derive(Ord, PartialOrd, Eq, PartialEq, Debug)]
-pub enum DeleteSumParticipant {
-    Ok,
-    DoesNotExist,
+#[derive(From)]
+pub(crate) struct LocalSeedDictWrite<'a>(&'a LocalSeedDict);
+
+impl ToRedisArgs for LocalSeedDictWrite<'_> {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        let args: Vec<(PublicSigningKeyWrite, EncryptedMaskSeedWrite)> = self
+            .0
+            .iter()
+            .map(|(pk, seed)| {
+                (
+                    PublicSigningKeyWrite::from(pk),
+                    EncryptedMaskSeedWrite::from(seed),
+                )
+            })
+            .collect();
+
+        args.write_redis_args(out)
+    }
 }
 
-impl FromRedisValue for DeleteSumParticipant {
-    fn from_redis_value(v: &Value) -> RedisResult<DeleteSumParticipant> {
+impl<'a> ToRedisArgs for &'a LocalSeedDictWrite<'a> {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        let args: Vec<(PublicSigningKeyWrite, EncryptedMaskSeedWrite)> = self
```

Comment:
Do we actually need the type annotation here? Seems like the `(PublicSigningKeyWrite::from(pk),EncryptedMaskSeedWrite::from(seed))` in the `map` should be enought?

Reply:
Unfortunately yes, otherwise I get an error at `args.write_redis_args(out)`
But I can write
```rust
            .map(|(pk, seed)| {
                (
                    From::from(pk),
                    From::from(seed),
                )
            })
```
instead of 
```rust
            .map(|(pk, seed)| {
                (
                    PublicSigningKeyWrite::from(pk),
                    EncryptedMaskSeedWrite::from(seed),
                )
            })
```
to make it a bit shorter